app/assets/javascripts/components/projects/SideBar/index.js

```diff
@@ -1,40 +1,70 @@
-import React from 'react';
-import SideBarButton from './SideBarButton';
-import PropTypes from 'prop-types';
-import SideBarButtonShape from '../../shapes/sideBarButton';
-import classname from 'classnames';
+import React from "react";
+import SideBarButton from "./SideBarButton";
+import PropTypes from "prop-types";
+import classname from "classnames";
 
-const SideBar = ({
-  buttons
-}) =>
-  <div className="SideBar">
-    <ul>
-      {
-        buttons.map(button => {
+const SideBar = ({ reverse, visibleColumns, toggleColumn, reverseColumns }) => {
+  const buttons = [
+    {
+      description: I18n.t("revert_columns_tooltip"),
+      onClick: reverseColumns,
+      "data-id": "reverse-button",
+      isVisible: reverse,
+      icon: "fas fa-columns"
+    },
+    {
+      description: I18n.t("toggle_column", { column: "chilly bin" }),
+      onClick: () => toggleColumn("chillyBin"),
+      "data-id": "toggle-chilly-bin",
+      isVisible: visibleColumns.chillyBin,
+      icon: "fas fa-snowflake"
+    },
+    {
+      description: I18n.t("toggle_column", { column: "backlog" }),
+      onClick: () => toggleColumn("backlog"),
+      "data-id": "toggle-backlog",
+      isVisible: visibleColumns.backlog,
+      icon: "fas fa-th-list"
+    },
+    {
+      description: I18n.t("toggle_column", { column: "done" }),
+      onClick: () => toggleColumn("done"),
+      "data-id": "toggle-done",
+      isVisible: visibleColumns.done,
+      icon: "fas fa-check-circle"
+    }
+  ];
```

Comment:
If you want to you can use `useMemo` here:
```js
const buttons = useMemo(() => [
  {
    description: I18n.t("revert_columns_tooltip"),
    onClick: reverseColumns,
    "data-id": "reverse-button",
    isVisible: reverse,
    icon: "fas fa-columns"
  },
  {
    description: I18n.t("toggle_column", { column: "chilly bin" }),
    onClick: () => toggleColumn("chillyBin"),
    "data-id": "toggle-chilly-bin",
    isVisible: visibleColumns.chillyBin,
    icon: "fas fa-snowflake"
  },
  {
    description: I18n.t("toggle_column", { column: "backlog" }),
    onClick: () => toggleColumn("backlog"),
    "data-id": "toggle-backlog",
    isVisible: visibleColumns.backlog,
    icon: "fas fa-th-list"
  },
  {
    description: I18n.t("toggle_column", { column: "done" }),
    onClick: () => toggleColumn("done"),
    "data-id": "toggle-done",
    isVisible: visibleColumns.done,
    icon: "fas fa-check-circle"
  }
], [reverse, visibleColumns, toggleColumn, reverseColumns]);
```

This way, this array will only be recreated when the component is created or some of the dependencies (`reverse`, `visibleColumns`, `toggleColumn`, `reverseColumns`) changed. 😄

Reply:
Nice, I will change this, thank you!